### PR TITLE
enh(driver):rest support cloud token

### PR DESCRIPTION
**Pull request by xleili**

### CODE REVIEW

File: typescript-rest/example/cloud_example.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 
-options.host = '127.0.0.1';
-//your cloud token
-options.query = {token:'c37ef4dbec8708c0227b4e8cb84ffffb9b8711a1'}
-options.port = 6041
+// add your cloud url
+options.cloudUri ='192.168.1.163:8085'
 
+// your cloud token
+options.query = {token:'c37ef4dbec8708c0227b4e8cb84ffffb9b8711a1'}
```
Comment (sangshuduo) [inline typescript-rest/example/cloud_example.ts:7]: should use environment variable

File: typescript-rest/src/cursor.ts [hunk 2]
```diff
@@ -14,10 +14,11 @@ export class TDengineCursor {
...
             host: options.host,
             path: options.path,
+            cloudUri: options.cloudUri,
             port: options.port,
-            schema:options.schema,
-            query:options.query,
-            fragment:options.fragment,
+            schema: options.schema,
+            query: options.query,
+            fragment: options.fragment,
```
Comment (sangshuduo) [inline typescript-rest/src/cursor.ts:21]: extra space

File: typescript-rest/src/options.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Agent } from "http";
-import { Headers } from "node-fetch";
+// import { Headers } from "node-fetch";
```
Comment (sangshuduo) [inline typescript-rest/src/options.ts:2]: should remove unused code instead of comment off

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

File: typescript-rest/example/cloud_example.ts (hunk 1 above)
Comment (xleili) [reply typescript-rest/example/cloud_example.ts:7]: fixed

File: typescript-rest/src/options.ts (hunk 3 above)
Comment (xleili) [reply typescript-rest/src/options.ts:2]: removed

File: typescript-rest/src/request.ts [hunk 4]
```diff
@@ -1,53 +1,71 @@
...
-    constructor(url: Uri, user:User) {
+    constructor(url: Uri, user: User) {
         this.uri = url;
         this.user = user;
-        this.options = {
-            method: 'POST',
-            body:'',
-            headers:{'Authorization':this._token()}
-        }
+        if(url.cloudUri==null ||url.cloudUri ==undefined ){
```
[the remark this answers is not in the document]
Comment (xleili) [reply typescript-rest/src/request.ts:13]: reformated

File: typescript-rest/src/cursor.ts (hunk 2 above)
Comment (xleili) [reply typescript-rest/src/cursor.ts:21]: this space is added atuomatically after using VS code "Format Document" option.

File: typescript-rest/index.ts [hunk 5]
```diff
@@ -1,14 +1,11 @@
 
 import {TDConnect,Options} from './src/connect';
 let options:Options = {
-    host: '127.0.0.1',
-    port: 6041,
     path: '/rest/sqlutc',
-    user: 'root',
-    passwd: 'taosdata',
-    schema: 'http'
+    schema: 'http',
```
Comment (zitsen) [inline typescript-rest/index.ts:5]: For URL, better use `scheme` imo.

**Review by zitsen — APPROVED** [review]
LGTM, one comment.

**Review by sangshuduo — APPROVED** [review]
LGTM

File: typescript-rest/index.ts (hunk 5 above)
Comment (xleili) [reply typescript-rest/index.ts:5]: fixed

**Pull request merged**